src/main/java/com/blackducksoftware/integration/jira/web/action/MigrationActions.java

```diff
@@ -0,0 +1,126 @@
+    private static final String TIME_FORMAT = "yyyy-MM-dd HH:mm:ss '(UTC)'";
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final SchedulerService schedulerService;
+    private final JiraSettingsAccessor jiraSettingsAccessor;
+    private final MigrationAccessor migrationAccessor;
+    private final AlertMigrationRunner alertMigrationRunner;
+
+    public MigrationActions(SchedulerService schedulerService, JiraSettingsAccessor jiraSettingsAccessor, MigrationAccessor migrationAccessor) {
+        this.schedulerService = schedulerService;
+        this.jiraSettingsAccessor = jiraSettingsAccessor;
+        this.alertMigrationRunner = new AlertMigrationRunner(jiraSettingsAccessor);
+        this.migrationAccessor = migrationAccessor;
+        schedulerService.registerJobRunner(MIGRATION_JOB_RUNNER_KEY, alertMigrationRunner);
+    }
+
+    public void removeProjectsFromCompletedList(List<String> projectsToDelete) {
+        List<String> migratedProjects = migrationAccessor.getMigratedProjects();
+        List<String> updatedProjects = migratedProjects.stream()
+                                           .filter(project -> !projectsToDelete.contains(project))
+                                           .collect(Collectors.toList());
+        migrationAccessor.updateMigratedProjects(updatedProjects);
+    }
+
+    public void startMigration() throws SchedulerServiceException {
+        JobRunnerKey jobRunnerKey = JobRunnerKey.of(MIGRATION_JOB_NAME);
+        JobConfig jobConfig = JobConfig
+                                  .forJobRunnerKey(jobRunnerKey)
+                                  .withRunMode(RunMode.RUN_LOCALLY)
+                                  .withSchedule(Schedule.runOnce(Date.from(Instant.now())));
+
+        JobId jobId = JobId.of(MIGRATION_JOB_NAME);
+        schedulerService.scheduleJob(jobId, jobConfig);
+        logger.info(String.format("%s scheduled to run now.", AlertMigrationRunner.HUMAN_READABLE_TASK_NAME));
+    }
+
+    public MigrationDetails getMigrationDetails() {
+        ProjectMappingConfigModel projectMapping = jiraSettingsAccessor.createGlobalConfigurationAccessor().getIssueCreationConfig().getProjectMapping();
+        BlackDuckJiraConfigSerializable config = new BlackDuckJiraConfigSerializable();
+        config.setHubProjectMappingsJson(projectMapping.getMappingsJson());
+
+        List<JiraProject> jiraProjects = config.getHubProjectMappings().stream().map(mapping -> mapping.getJiraProject()).collect(Collectors.toList());
+        List<String> projectsToBeMigrated = new ArrayList<>();
+
+        List<String> migratedProjects = migrationAccessor.getMigratedProjects();
+        if (null != jiraProjects && !jiraProjects.isEmpty()) {
+            jiraProjects.stream()
+                .filter(jiraProject -> !migratedProjects.contains(jiraProject.getProjectName()))
+                .forEach(project -> projectsToBeMigrated.add(project.getProjectName()));
```

Comment:
You can instead set projectsToBeMigrated equal to this stream manipulation. it would be something similar to

`projectToBeMigrated = jiraProjects.stream().map(JiraProject::getProjectName).filter(projectName -> !migratedProjects.contains(projectName).collect(Collectors.toList());`